lib/nx_zlib.c

```diff
@@ -1025,10 +1025,11 @@ void nx_hw_init(void)
 
 	if(type_selector != NULL){
 		nx_config.gzip_selector = str_to_num(type_selector);
-		prt("gzip_selector: %d (1-SW;2-NX;3/4-MIX)\n", nx_config.gzip_selector);
-		if(nx_config.gzip_selector == 0 || nx_config.gzip_selector > 4) {
-			prt("Unrecognized option, defaulting to NX.\n");
-			nx_config.gzip_selector = GZIP_NX;
+		prt("gzip_selector: %d (0-AUTO;1-SW;2-NX;3/4-MIX)\n", nx_config.gzip_selector);
+		/* check for misconfiguration */
+		if(nx_config.gzip_selector > 4) {
+			prt("Unrecognized option, defaulting to AUTO.\n");
```

Comment:
These lines were only added in the second commit, but should actually be part of the first one, which sets AUTO as the default.

Reply:
Agreed, I may have splitted this by mistake in the many rebases I did.